sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/PartitionReceiverLiveTests.cs

```diff
@@ -36,31 +41,1340 @@ public class PartitionReceiverLiveTests
+                    Assert.That(cancellationSource.IsCancellationRequested, Is.False, "The cancellation token should not have been signaled.");
+
+                    cancellationSource.Cancel();
+                    await exclusiveMonitor.ReadTask;
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="PartitionReceiver" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ReceiverCannotReadWithLowerOwnerLevelThanActiveReader()
+        {
+            await using (EventHubScope scope = await EventHubScope.CreateAsync(1))
+            {
+                using var cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(TimeSpan.FromMinutes(3));
+
+                var higherOptions = new PartitionReceiverOptions { OwnerLevel = 40 };
+                var lowerOptions = new PartitionReceiverOptions { OwnerLevel = 20 };
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+                var sourceEvents = EventGenerator.CreateEvents(200).ToList();
+
+                var partition = (await QueryPartitionsAsync(connectionString, cancellationSource.Token)).First();
+                await SendEventsAsync(connectionString, sourceEvents, new CreateBatchOptions { PartitionId = partition }, cancellationSource.Token);
+
+                await using (var higherReceiver = new PartitionReceiver(EventHubConsumerClient.DefaultConsumerGroupName, partition, EventPosition.Earliest, connectionString, higherOptions))
+                await using (var lowerReceiver = new PartitionReceiver(EventHubConsumerClient.DefaultConsumerGroupName, partition, EventPosition.Earliest, connectionString, lowerOptions))
+                {
+                    var higherMonitor = MonitorReadingEvents(higherReceiver, int.MaxValue, cancellationSource.Token);
+                    await Task.WhenAny(higherMonitor.StartCompletion.Task, Task.Delay(Timeout.Infinite, cancellationSource.Token));
+                    Assert.That(cancellationSource.IsCancellationRequested, Is.False, "The cancellation token should not have been signaled.");
+
+                    var lowerReadTask = ReadEventsAsync(lowerReceiver, int.MaxValue, cancellationSource.Token);
+                    Assert.That(async () => await lowerReadTask, Throws.InstanceOf<EventHubsException>().And.Property(nameof(EventHubsException.Reason)).EqualTo(EventHubsException.FailureReason.ConsumerDisconnected), "The lower-level read should be rejected.");
+                    Assert.That(cancellationSource.IsCancellationRequested, Is.False, "The cancellation token should not have been signaled.");
+
+                    cancellationSource.Cancel();
+                    await higherMonitor.ReadTask;
```

Comment:
Should we assert on the event count from this `ReadTask`?

Reply:
We could, and it wouldn't hurt anything... but I saw it as orthogonal to the purpose of the test itself - which was to validate that the lower reader gets rejected.   We've got other tests that validate readers with a higher owner level or exclusive when others are non-exclusive can read without being interrupted, so it feels redundant here.